news/2017-01-04-tip4-testing.asciidoc

```diff
@@ -0,0 +1,145 @@
+---
+layout: news
+title: Weld Tip 4 - Testing Weld apps
+author: Tomas Remes, Matej Novotny
+priority: 1
+change_frequency: weekly
+excerpt: Taking a dive into CDI/Weld application testing.
+desc: Weld Tip 4 - Testing Weld apps
+tags: [tips]
+---
+:linkattrs:
+:toc: macro
+:toc-title: Contents
+:icons: font
+
+toc::[]
+
+In this follow-up article of our Weld Tips series we are going to focus on integration testing of CDI/Weld applications.
+To set things straight - by integration testing we mean testing in some kind of container and hence in certain environment (SE, servlet, full EE server).
+However, no matter the environment we always make use of link:http://arquillian.org/[Arquillian] integration testing framework.
+
+== Arquillian as the chosen framework for testing
+While Arquillian may look complex at first, it is certainly worth it. Here are some of the advantages it gives you:
+
+* Allows usage of CDI `@javax.inject.Inject` directly in test classes
+* Works well with ShrinkWrap making archive creation and deployment a breeze
+** link:http://arquillian.org/guides/shrinkwrap_introduction/[ShrinkWrap] can create any type of deployment archive (WAR, JAR, even infamous EAR) with exactly the capabilities you wish it to have
+** You can use dedicated `beans.xml` or `@Alternatives` for given test case
+* Portability - allows to write one test and run it on any container
+** Arquillian only needs correct adapter; therefore, combined with Maven profiles this gives you some nice fire power for matrix testing
+* Allows for very complex testing scenarios
+** For instance you can create several deployments which you then manually (if you so wish) deploy/undeploy
+** This makes it viable even for cluster testing (session replication, failover scenarios, ...)
+* Can start the application server itself or can just connect to running one
+
+In fact writing tests with Arquillian is quite straightforward. It's basically about right usage of right annotations. Basic test could like this:
+[ source, java ]
+ ----
```

Comment:
Remove the leading space, otherwise the code is badly formatted. The same applies to L69.